From 3918800b50b08f0fc3b61b9aaf6446462253c25d Mon Sep 17 00:00:00 2001
From: Jared McArthur <j-mcarthur@ti.com>
Date: Mon, 29 Jul 2024 11:28:39 -0500
Subject: [PATCH] arm64: dts: ti: k3-am62p: Add gpio-reserved-ranges for
 main_gpio1

Commit 29075cc09f43 ("arm64: dts: ti: Introduce AM62P5 family of SoCs")
introduced the main_gpio1 node and included the ngpios, but missed the
gpio-reserved-ranges property. As a result, the driver could try to
access gpios that do not exist. Fix this by introducing the
gpio-reserved-ranges property.

Fixes: 29075cc09f43 ("arm64: dts: ti: Introduce AM62P5 family of SoCs")
Signed-off-by: Jared McArthur <j-mcarthur@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62p-main.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-main.dtsi
index 57383bd2eaeb..3bf0fab7eb91 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-main.dtsi
@@ -65,4 +65,5 @@ &main_gpio1 {
 	gpio-ranges = <&main_pmx0 0 94 32>, <&main_pmx0 42 137 5>,
 			<&main_pmx0 47 143 3>, <&main_pmx0 50 149 2>;
 	ti,ngpio = <52>;
+	gpio-reserved-ranges = <32 10>;
 };
-- 
2.34.1

